static/js/subs.js

```diff
@@ -366,6 +374,10 @@ exports.mark_sub_unsubscribed = function (sub) {
             settings.collapse('hide');
         }
 
+        // Update subscribers count
+        var count = stream_data.get_subscribers_count(sub.name) - 1;
```

Comment:
Why do you need a `-1` here?  Since the user was marked as no longer subscribed above, this function should be producing correct output.

I think maybe the issue here is that `get_subscribers_count` shouldn't be computing `length`, it should instead be counting the elements that have `.subscribed=true`...